src/db/collections/status.rs

```diff
@@ -24,14 +25,43 @@ impl MongoDb {
             .collection::<StatusDocument>(StatusDocument::COLLECTION)
             .find_one(doc! {}, None)
             .await
-            .map(|doc| doc.map(|doc| doc.network_name))
+            .map(|doc| doc.and_then(|doc| doc.network_name))
     }
 
     /// Sets the name of the network.
     pub async fn set_network_name(&self, network_name: String) -> Result<(), Error> {
         self.0
             .collection::<StatusDocument>(StatusDocument::COLLECTION)
-            .insert_one(StatusDocument { network_name }, None)
+            .update_one(
+                doc! {},
+                doc! { "$set": { "network_name": network_name } },
+                UpdateOptions::builder().upsert(true).build(),
+            )
+            .await?;
+
+        Ok(())
+    }
+
+    /// Get the current ledger index.
+    pub async fn get_ledger_index(&self) -> Result<Option<MilestoneIndex>, Error> {
+        self.0
+            .collection::<StatusDocument>(StatusDocument::COLLECTION)
+            .find_one(doc! {}, None)
+            .await
+            .map(|doc| doc.and_then(|doc| doc.ledger_index))
+    }
+
+    /// Sets the current ledger index.
+    pub async fn set_ledger_index(&self, ledger_index: MilestoneIndex) -> Result<(), Error> {
```

Comment:
This is - admittedly - a bit nit-picky, but `set_` to me implies it will actually be set to the given value which is not the case. A better name might be `update_ledger_index`. The doc for this method should probably also explain that the ledger index gets updated only if it marks a new max.

Reply:
Fair